RFC: Remove command-line Git from Mac and Windows binaries

tkelman opened the pull request:
This is step 4 to fix https://github.com/JuliaLang/julia/issues/13958#issuecomment-226922219

Git.jl is mostly set up at https://github.com/JuliaPackaging/Git.jl (thanks @ararslan for your help cleaning it up), though still need to
- [x] handle installing Git for you when a working version isn't found on the path (https://github.com/JuliaPackaging/Git.jl/pull/7)
- [x] register the package (https://github.com/JuliaLang/METADATA.jl/pull/5589)

Then
- [x] test that binaries built from this branch work properly and won't break appveyor, coverage (https://github.com/JuliaCI/Coverage.jl/pull/108), etc. If CI passes I'm satisfied on this one, though it may need a little bit of cleanup during the RC period

Some of base Julia's tests assume coreutils (and perl) are present, and on Windows we've been borrowing those from the bundled Git.
- [x] reinstate busybox-w32 for these ~~when GPL is allowed, skip the tests in a no-GPL build if coreutils isn't found~~. Or powershell-ify the tests.

tkelman committed:
Remove command-line Git from Mac and Windows binaries

tkelman commented:
I expect to finish up the rest of the test modifications here tonight, just so people are aware.

StefanKarpinski commented:
I think that @vtjnash was working on adding BusyBox back. Not sure how that went.

tkelman commented:
It's not like we need to build it from source which might get a bit complicated, I was just planning on downloading a binary in win-extras since we'll only be using it for test purposes for the time being. We could leverage it for more interesting things later if we want to, but only in GPL-allowed builds.

tkelman committed:
Reinstate downloading of busybox-w32 in make win-extras

We may want to make this conditional on USE_GPL_LIBS, though since
we're only running it as a command line executable and not linking
to it, it probably falls under the aggregation clause

tkelman committed:
Use busybox-w32 in spawn and cmdlineargs tests on Windows

note that one test involving perl is being skipped, but it
had been disabled until recently - ref e412c69a6e77b0751e0fa0303633e52154937a46

tkelman commented on test/spawn.jl:
```diff
@@ -4,34 +4,31 @@
-#  - GNU coreutils (or equivalent)
-#  - perl
-
-#TODO:
```
yay addressing 4-year-old TODO's https://github.com/JuliaLang/julia/commit/1ddd1639368fe3d1ec793e5287bc9eff1f31f79a

tkelman commented on test/spawn.jl:
```diff
@@ -53,28 +50,30 @@ if valgrind_off
-prefixer(prefix, sleep) = `perl -nle '$|=1; print "'$prefix' ", $_; sleep '$sleep';'`
-@test success(pipeline(`perl -le '$|=1; for(0..2){ print; sleep 1 }'`,
+if is_unix()
+    prefixer(prefix, sleep) = `perl -nle '$|=1; print "'$prefix' ", $_; sleep '$sleep';'`
```
can someone explain what this perl is doing? it's been disabled for a long time, ref https://github.com/JuliaLang/julia/commit/e412c69a6e77b0751e0fa0303633e52154937a46, I was the one who turned it back on recently in 373db0f7b510403fd5e22431a830df9be20620ed

tkelman committed:
fall back to using coreutils if busybox is not present

it usually won't be in source builds

tkelman commented:
osx travis failure was a libgit2 segfault probably caused by https://github.com/JuliaLang/julia/pull/16537
